packages/api/src/utils/logs.ts

```diff
@@ -0,0 +1,310 @@
+  public currentTs: number
+  public _finished = false
+  public metadata: any = {}
+
+  constructor (request: any, ctx: ExecutionContext, opts: ILoggingOptions) {
+    this.ctx = ctx
+    this.opts = opts
+    this._times = new Map()
+    this.currentTs = this.startTs
+
+    const cf = request.cf
+    let rCf
+    /* eslint-disable-next-line */
+    if (cf) {
+      const { tlsClientAuth, tlsExportedAuthenticator, ...rest } = cf
+      rCf = rest
+    }
+    this.metadata = {
+      user: {
+        id: 0
+      },
+      request: {
+        url: request.url,
+        method: request.method,
+        headers: buildMetadataFromHeaders(request.headers),
+        cf: rCf
+      },
+      cloudflare_worker: {
+        version: opts.version,
+        commit: opts.commithash,
+        branch: opts.branch,
+        worker_id: nanoid(10),
+        worker_started: this.startTs
+      }
+    }
+
+    // As this class must be instantiated once per request, we can automatically
+    // start the timing of the request here
+    this.time('request')
+  }
+
+  _date () {
+    const now = Date.now()
+    if (now === this.currentTs) {
+      const dt = new Date().toISOString()
+      /**
+       * Fake increment the datetime string to order the logs entries
+       * It won't leap seconds but for most cases it will increment by 1 the datetime milliseconds
+       */
+      const newDt = dt.replace(/\.(\d*)Z/, (s, p1, p2) => {
+        return `.${String(Number(p1) + this.logEventsBatch.length)}Z`
+      })
+      return new Date(newDt).toISOString()
+    } else {
+      this.currentTs = now
+      return new Date().toISOString()
+    }
+  }
+
+  /**
+   * Add log entry to batch
+   *
+   * @param {any} body
+   */
+  _add (body: any) {
+    this.logEventsBatch.push(body)
+  }
+
+  async postBatch () {
+    if (this.logEventsBatch.length > 0) {
+      const batchInFlight = [...this.logEventsBatch]
+      this.logEventsBatch = []
+      const rHost: string = batchInFlight[0].metadata.request.headers.host
+      const body = JSON.stringify(batchInFlight)
+      const request = {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.opts?.token ?? ''}`,
+          'Content-Type': 'application/json',
+          'User-Agent': `Cloudflare Worker via ${rHost}`
+        },
+        body
+      }
+      const resp = await fetch(logtailApiURL, request)
+      if (this.opts?.debug != null) {
+        console.info(
+          `[${this._date()}]`,
+          `${batchInFlight.length} logs pushed with status ${resp.status}.`
+        )
+      }
+    }
+  }
+
+  /**
+   * End instance, push logs and servers timings
+   *
+   * @param {Response} response
+   */
+  async end (response: Response) {
+    if (this._finished) {
+      throw new Error(
+        `end() has already been called on this Logging instance.
+        You must make a new instance per request.`
+      )
+    }
+    this._finished = true
+    if (this.opts?.debug != null) {
```

Comment:
Is this comparison deliberate? `opts.debug` is a boolean, and both `true != null` and `false != null` will both be true.

Reply:
Good spot, this was an incorrect comparison. Updated to `?? false` as per https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md